app/services/katello/pulp3/content_view_version/importable_repositories.rb

```diff
@@ -2,63 +2,115 @@ module Katello
+          # Set up a 2 different list of importable root repositories
           # creatable: repos that are part of the metadata but not in the library.
           #         They are ready to be created
           # updatable: repo that are both in the metadata and library.
           #         These may contain updates to the repo and hence ready to be updated.
-          metadata_map = Import.metadata_map(metadata, custom_only: !redhat, redhat_only: redhat)
-          metadata_map.keys.each do |product_label, repo_label|
-            product = Katello::Product.in_org(organization).find_by(label: product_label)
-            fail _("Unable to find product '%s' in organization '%s'" % [product_label, organization.name]) if product.blank?
-            params = metadata_map[[product_label, repo_label]]
-            if params[:gpg_key].blank?
-              params[:gpg_key_id] = nil
-            else
-              params[:gpg_key_id] = organization.gpg_keys.find_by(name: params[:gpg_key][:name]).id
-            end
-            content = params[:content]
-            params = params.except(:redhat, :product, :gpg_key, :content)
-            if repositories_in_library.include? [product_label, repo_label]
-              repo = ::Katello::RootRepository.find_by(product: product, label: repo_label)
-              updatable << { repository: repo, options: params.except(:label, :name, :content_type) }
-            elsif redhat
-              product_content = product_content_by_label(content[:label])
+          @metadata_repositories.each do |repo|
+            product = product_for_metadata_repo(repo)
+            fail _("Unable to find product '%s' in organization '%s'" % [repo.product.label, @organization.name]) if product.blank?
+
+            # TODO: is it valid to look up the root this way?
+            root = product.root_repositories.find { |r| r.label == repo.label }
+            #if product_repo_labels.include?([repo.product.label, repo.label])
+            if root
+              #root = ::Katello::RootRepository.find_by(product: product, label: repo.label)
+              updatable << { repository: root, options: update_repo_params(repo) }
+            elsif repo.redhat
+              content = repo.content
+              product_content = product_content_by_label(content.label)
               substitutions = {
-                basearch: params[:arch],
-                releasever: params[:minor]
+                basearch: repo.arch,
+                releasever: repo.minor
               }
               creatable << { product: product, content: product_content, substitutions: substitutions }
             else
-              creatable << { repository: product.add_repo(params) }
+              creatable << { repository: product.add_repo(create_repo_params(repo)) }
             end
           end
         end
+
+        private
+
+        def product_for_metadata_repo(repo)
+          if repo.redhat && repo.product.cp_id
```

Comment:
what happens if it is a custom repo with the same name as a redhat repo ?

Reply:
I guess it'll be treated as a custom repo! Hopefully no one is doing that